wagtail/models/sites.py

```diff
@@ -41,7 +46,7 @@ def get_site_for_hostname(hostname, port):
         )
         .filter(Q(hostname=hostname) | Q(is_default_site=True))
         .order_by("match")
-        .select_related("root_page")
+        .select_related("root_page")[:3]
```

Comment:
Note: There's a slight issue with this implementation. Whilst we invalidate the cache on _Site_ changes, we don't on _Page_ changes, meaning a potentially stale version of the root page may be provided.

Reply:
This is a valid scenario which needs covering.

(aside: tests are failing)

Reply:
As it turns out, I already did :facepalm:  There's a signal handler to specifically clear the cache when this happens.